core/ide/src/main/java/org/alice/ide/x/AstI18nFactory.java

```diff
@@ -368,24 +355,24 @@ protected SwingComponentView<?> createPropertyComponent(InstanceProperty<?> prop
       }
     } else {
       rv = null;
-      if (property instanceof NodeProperty<?>) {
-        if (property instanceof ExpressionProperty) {
-          rv = this.createExpressionPropertyPane((ExpressionProperty) property);
+      if (property instanceof NodeProperty<?> nodeProperty) {
+        if (property instanceof ExpressionProperty expressionProperty) {
+          rv = this.createExpressionPropertyPane(expressionProperty);
         } else {
-          rv = this.createGenericNodePropertyPane((NodeProperty<?>) property);
+          rv = this.createGenericNodePropertyPane(nodeProperty);
         }
-      } else if (property instanceof ResourceProperty) {
-        rv = this.createResourcePropertyPane((ResourceProperty) property);
+      } else if (property instanceof ResourceProperty resourceProperty) {
+        rv = this.createResourcePropertyPane(resourceProperty);
       } else if (property instanceof ListProperty<?>) {
         if (property instanceof NodeListProperty<?>) {
-          if (property instanceof StatementListProperty) {
-            rv = this.createStatementListPropertyPane((StatementListProperty) property);
-          } else if (property instanceof SimpleArgumentListProperty) {
-            rv = this.createSimpleArgumentListPropertyPane((SimpleArgumentListProperty) property);
-          } else if (property instanceof KeyedArgumentListProperty) {
-            rv = this.createKeyedArgumentListPropertyPane((KeyedArgumentListProperty) property);
-          } else if (property instanceof ExpressionListProperty) {
-            rv = this.createExpressionListPropertyPane((ExpressionListProperty) property);
```

Comment:
we got so far through this file before we hit numbers that i  thought i was safe